src/lib/FetchingProvider.tsx

```diff
@@ -77,22 +85,28 @@ export function FetchingProvider(props: FetchingProviderApi) {
 
   React.useEffect(() => {
     let isStillMounted = true;
+    const initEmptyTranslationMap = () => {
+      logTranslationsNOK();
+      setState({ translations: {}, isFetched: true });
+    };
 
     setState((state: State) => ({ ...state, isFetched: false }));
     onFetchingStart();
 
     fetch(url)
       .then(r => r.json())
-      .then(response => {
+      .then(rawTranslations => transform(rawTranslations))
+      .then(transformedTranslations => identityWithFalsenessCheck(transformedTranslations))
+      .then(checkedTranslations => {
         if (isStillMounted) {
           setState({
-            translations: transform(response),
+            translations: checkedTranslations,
             isFetched: true,
           });
           onFetchingEnd();
         }
       })
-      .catch(onFetchingError);
+      .catch(e => (onFetchingError !== noop ? onFetchingError(e) : initEmptyTranslationMap()));
```

Comment:
Sure, i agree with the first point. I went with this approach since i wanted to get the logs in and it looked clearer that way, now that we are removing the logs, i'll clean it up.

I don't think there is anything we can do other than just get the previous state ( which will be {} most of the times ), when we get any type of error.

A quick note, since we are defaulting to the previous state on network failure, why not just default to the previous state when a faulty translation map is given as well ( which again, will just be {} most of the time )? I'll write it up like that and i'll revert in case i missed something.